### PR TITLE
Api 18149 update v2 enum

### PR DESCRIPTION
**Pull request by stiehlrod**
## Description of change
Changes the status attribute returned by /claims and /claims/{id} in our v2 services to an enum value, similar to our v1 services. 

## Original issue(s)
[API-18149](https://vajira.max.gov/browse/API-18149)

### CODE REVIEW

File: modules/claims_api/lib/claims_api/evss_status_mapper.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# Maps a given 'status' to a known enum status
+
+module ClaimsApi
+  class EvssStatusMapper
```
Comment (jw81) [inline modules/claims_api/lib/claims_api/evss_status_mapper.rb:6]: The upstream service we're calling to get these values is actually BGS (not EVSS).  So we should prob rename this class to something like `BGSClaimStatusMapper`

File: spec/support/schemas/claims_api/v2/veterans/claims/claims.json [hunk 2]
```diff
@@ -53,7 +53,15 @@
         "example": "600131328"
       },
       "status": {
-        "type": "string",
+        "type": { "enum": 
+        [
```
Comment (jw81) [inline spec/support/schemas/claims_api/v2/veterans/claims/claims.json:57]: Looks like we missed adding `PENDING` here.

File: spec/support/schemas/claims_api/v2/veterans/claims/claim.json [hunk 3]
```diff
@@ -61,7 +61,15 @@
       "example": "600131328"
     },
     "status": {
-      "type": "string",
+      "type": { "enum": 
+        [
```
Comment (jw81) [inline spec/support/schemas/claims_api/v2/veterans/claims/claim.json:65]: Looks like we missed adding `PENDING` here.

**Comment by jw81** [comment]
Now that the V2 JSON schemas were updated, we'll need to run `rswag` so we can get an updated OAS/developer docs.

File: modules/claims_api/app/models/claims_api/evss_claim.rb [hunk 4]
```diff
@@ -10,6 +10,17 @@ class EVSSClaim
     # - https://github.com/department-of-veterans-affairs/vets-website/blob/fcf944bd4319684f5eb3d1901606801d01a9a55e/src/applications/claims-status/utils/helpers.js#L9
     # As of the time of this comment they exist as the source of truth for this information
     # in the future we'll likely try to make this the source of truth
+    #
+    # 8/4/2022
```
Comment (acovrig) [inline modules/claims_api/app/models/claims_api/evss_claim.rb:14]: I like this comment block

**Review by acovrig — APPROVED** [review]

**Review by jw81 — APPROVED** [review]

**Pull request merged**